src/librustc_typeck/diagnostics.rs

```diff
@@ -1533,6 +1533,26 @@ For information on the design of the orphan rules, see [RFC 1023].
 [RFC 1023]: https://github.com/rust-lang/rfcs/pull/1023
 "##,
 
+E0118: r##"
+Rust can't find a base type for an implementation you are providing, or the type
+cannot have an implementation. For example, a typedef can't have an
+implementation, since it isn't its own type (this was done in PR #6087):
```

Comment:
This text isn't quite accurate. Whether something is a type alias is not relevant. See for example [this code on play](http://is.gd/zZXfO9). The rule is that you can only do an inherent impl (meaning, an impl without a trait) for a named type (struct, enum, or trait) defined in the local crate. The reason that your example below is prohibited is because the type `[char, ..9]` is an array type, which is not a named type.